### PR TITLE
APM-4483 MTLS changes

### PR DESCRIPTION
**Pull request by acarriedev**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/proxygen.yaml [hunk 1]
```diff
@@ -256,25 +253,84 @@ info:
     ```
     securitySchemes:  # limit this to only those used in your API
       nhs-login-p0:
-      $ref: https://proxygen.ptl.api.platform.nhs.uk/components/securitySchemes/nhs-login-p0
+      $ref: https://proxygen.prod.api.platform.nhs.uk/components/securitySchemes/nhs-login-p0
```
Comment (acarriedev) [inline specification/proxygen.yaml:256]: Changed all of these as they should be more stable/reliable in theory that ptl

File: specification/proxygen.yaml [hunk 2]
```diff
@@ -258,26 +257,138 @@ info:
...
     ```
     Each item references a publically available JSON description object, which is rendered into the final specification document.
 
-    ### Extensions
-    #### x-nhsd-apim
-    This custom extension to the OpenAPI specification describes the API back-end and is the primary source of information for infrastructure to be deployed by Proxy Generator API.
-    See the schema definition below for a full list of available/required attributes.
+    ### `x-nhsd-apim` Specification Extension
+
+    This [extension to the OpenAPI specification](https://swagger.io/docs/specification/openapi-extensions/) is read during API Instance deployment and controls functionality specific to the API Platform.
```
Comment (tonyheap) [inline specification/proxygen.yaml:278]: lower case "I" on "instance"?

File: specification/proxygen.yaml [hunk 3]
```diff
@@ -258,26 +257,138 @@ info:
...
+
+    #### Monitoring
+
+    The `monitoring` key controls whether the instance will be monitored by the API platform. This is enabled by default (set to `true`).
+
+    #### Target Specification
```
Comment (tonyheap) [inline specification/proxygen.yaml:284]: lower case S

File: specification/proxygen.yaml [hunk 4]
```diff
@@ -258,26 +257,138 @@ info:
...
+
+    The `target` key contains an object that defines the target (back-end) where your API proxy forwards requests.
+
+    ##### External Target
```
Comment (tonyheap) [inline specification/proxygen.yaml:288]: Lower case T

File: specification/proxygen.yaml [hunk 5]
```diff
@@ -258,26 +257,138 @@ info:
...
+
+    A service not hosted on the API Platform. (Contrast with Hosted Target, below.)
+
+    Two methods of securing the connection between the API proxy and target service are supported: API Key and mTLS.
+    
+    Both methods depend on a named secret (`target.secret` property) existing in the relevant environment. See [Create or update a secret](#put-/apis/-api_name-/environments/-environment-/secrets/-secret_name-).
+
+    ###### mTLS Security
```
Comment (tonyheap) [inline specification/proxygen.yaml:296]: Lower case S

File: specification/proxygen.yaml [hunk 6]
```diff
@@ -258,26 +257,138 @@ info:
...
+    target:
+      type: external
+      healthcheck: /_status
+      url: https://example.external.mtls.secured.backend.com
+      security:
+        type: mtls
+        secret: example-cert-key-pair-secret-name
+    ```
+
+    ###### API Key external target
```
Comment (tonyheap) [inline specification/proxygen.yaml:315]: lower case K - here and below

**Review by tonyheap — CHANGES_REQUESTED** [review]

File: specification/proxygen.yaml [hunk 7]
```diff
@@ -258,26 +257,138 @@ info:
...
+        - name: hello-world
+          image:
+            name: hello-world_hello-world
+            tag: 72537-sha6c0ccdb
+          environment:
+            LOG_LEVEL: info
+            UPSTREAM: 'https://internal-dev.api.service.nhs.uk'
+            NODE_ENV: production
+    ```
+    #### Rate Limiting
```
Comment (tonyheap) [inline specification/proxygen.yaml:357]: lower case L

File: specification/proxygen.yaml [hunk 8]
```diff
@@ -258,26 +257,138 @@ info:
...
+    ratelimiting:
+      proxy:
+        limit: 5
+        timeunit: minute
+      app-default:
+        limit: 10
+        timeunit: minute
+    ```
+    
+    #### API Products
```
Comment (tonyheap) [inline specification/proxygen.yaml:378]: lower case P

**Pull request merged**